### DISCUSSION TITLE
Self-hosted runners and services (not working)

### DISCUSSION THREAD

**Discussion by zaherg** [community/community#26527] [Actions]
<p>I have installed a self-hosted runner on my mac to try it, but it always failed to run at the point of initiating a service</p>
<p><img src="https://user-images.githubusercontent.com/27624/181098173-dd40e665-bb6f-4a4e-a3ac-7355ac2f147f.png" alt="Screen Shot 2019-11-06 at 2.55.16 PM.png" data-base62-sha1="fyfaMPJZsGbtVKUjLLOrFD8YcoO" width="562" height="249"></p>
<p><img src="https://user-images.githubusercontent.com/27624/181098174-ddcc67a8-1b74-46a7-ba2b-08a305bc5a79.png" alt="Screen Shot 2019-11-06 at 2.55.05 PM.png" data-base62-sha1="6X3bSrd0TZqAh3ONbd6sd74xodp" width="608" height="126"></p>
<p>Is this normal? or with self-hosted runners, I should also not sue services and try to use docker instead?</p>
<p>Thanks</p>

**Comment by zaherg** [comment]
<p>Any update about this one? I can’t seem to find any mention in the documentation that using services with self-hosted actions won’t work.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/linuxjuggler">@linuxjuggler</a>,</p>
<p>Thanks for being here and our sincere apologies for the delayed response, are you still having this issue?</p>

**Comment by zaherg** [comment]
<p>I didn’t try it anymore since no one confirmed that they have fixed it.</p>
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> Can you confirm it is working and fixed?</p>

**Comment by zaherg** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7d4af86668fea46f0526478b16f20adbd1b9d5b5.png" data-download-href="/uploads/short-url/hSoj1XdaP2zIVaR2WEFlJZqW3Qx.png?dl=1" title="Screen Shot 2019-12-29 at 5.12.45 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/27624/181098176-79278def-3961-4a2d-8b57-e0eca1d5bbd6.png" alt="Screen Shot 2019-12-29 at 5.12.45 PM.png" data-base62-sha1="hSoj1XdaP2zIVaR2WEFlJZqW3Qx" width="690" height="326" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7d4af86668fea46f0526478b16f20adbd1b9d5b5_2_690x326.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7d4af86668fea46f0526478b16f20adbd1b9d5b5_2_1035x489.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7d4af86668fea46f0526478b16f20adbd1b9d5b5_2_1380x652.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7d4af86668fea46f0526478b16f20adbd1b9d5b5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-12-29 at 5.12.45 PM.png</span><span class="informations">1894×895 87.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>From what I can tell this is not working, and the best option is to depend on using local docker (docker-compose or Dockerfile) instead of depending on the services with self-hosted version!</p>
<p>So to answer your question <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> : sadly, Yes I am still facing the same issue.</p>
<p><a href="https://github.com/linuxjuggler/local-actions/commit/d409b98a312fa2e576d7ce0ddf8a5bede722133a/checks?check_suite_id=377534803" rel="nofollow noopener">https://github.com/linuxjuggler/local-actions/commit/d409b98a312fa2e576d7ce0ddf8a5bede722133a/checks?check_suite_id=377534803</a></p>

**Comment by thboop** [comment]
<p>We currently only support container operations on Linux Runners.</p>
<p>Services, which use docker containers, will not work on OSX or Windows.</p>
<p>This error message is not particularly helpful, we have a PR in place to address the error message which has not been rolled out yet: <a href="https://github.com/actions/runner/pull/221" rel="nofollow noopener">https://github.com/actions/runner/pull/221</a></p>
<p>We have also passed along feedback to the team that supporting containers on non-Linux Runners would be helpful. You may also create an enhancement issue in the actions runner repo linked above.</p>

**Comment by zaherg** [comment]
<p>Thanks <a class="mention" href="/thboop">@thboop</a> for the reply.</p>
<p>Just to be clear, if I run the self hosted runner on Linux services should work without any problems right?</p>

**Comment by thboop** [comment, marked as answer]
<p><a class="mention" href="/linuxjuggler">@linuxjuggler</a> correct!</p>
<p>You will need to make sure docker is installed on the self hosted runner’s machine and in the path.</p>
<p>I was able to verify this was working earlier today, if you have access to a linux box, you can test this with the following workflow after setting up a self hosted runner:</p>
<pre><code>name: Test CI

on: [push, pull_request]

jobs:
  test:
    services:
      redis:
        image: redis
        ports:
          - 6379/tcp
    runs-on: ["self-hosted", "Linux"]
    steps:
    - run: docker ps
</code></pre>
<p>We can see the output from the step below, where the redis image is running in docker.</p>
<pre><code>2020-01-06T21:12:05.7169898Z CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES
2020-01-06T21:12:05.7171418Z 01d4c9dc666f redis "docker-entrypoint.sâ€¦" 1 second ago Up Less than a second 0.0.0.0:32773-&gt;6379/tcp 803c3ff666ca429781c69a13e9e8160c_redis_6dea08
2020-01-06T21:12:05.7358695Z Stop and remove container: 803c3ff666ca429781c69a13e9e8160c_redis_6dea08
</code></pre>

**Comment by zaherg** [comment]
<p>I’ll do thanks <a class="mention" href="/thboop">@thboop</a> </p>

**Comment by fgriberi** [comment]
<p>secrets are supported by self-hosted? </p>
